integration-tests/tests/multichain/actions/mod.rs

```diff
@@ -87,3 +101,293 @@ pub async fn single_signature_production(
+        signature.verify(&another_user_pk, &payload_hash_scalar);
+    assert!(is_signature_valid_for_user_pk);
+    assert_eq!(is_signature_valid_for_mpc_pk, false);
+    assert_eq!(is_signature_valid_for_another_user_pk, false);
+
+    // Check signature using ecdsa tooling
+    let k256_sig = k256::ecdsa::Signature::from_scalars(r, s).unwrap();
+    let user_pk_k256: k256::elliptic_curve::PublicKey<Secp256k1> =
+        k256::PublicKey::from_affine(user_pk).unwrap();
+
+    let ecdsa_local_verify_result = verify(
+        &k256::ecdsa::VerifyingKey::from(&user_pk_k256),
+        &payload_hash_reversed,
+        &k256_sig,
+    );
+    assert!(ecdsa_local_verify_result.is_ok());
+
+    // TODO: fix
+    // let ecdsa_signature: ecdsa::Signature<Secp256k1> =
+    //     ecdsa::Signature::from_scalars(r, s).unwrap();
+    // let ecdsa_verify_result = ecdsa::signature::Verifier::verify(
+    //     &k256::ecdsa::VerifyingKey::from(&user_pk_k256),
+    //     &payload_hash_reversed,
+    //     &ecdsa_signature,
+    // );
+    // assert!(ecdsa_verify_result.is_ok());
+    // let k256_verify_key = k256::ecdsa::VerifyingKey::from(&user_pk_k256);
+    // let k256_verify_result = k256_verify_key.verify(&payload_hash_reversed, &k256_sig);
+    // assert!(k256_verify_result.is_ok());
+
+    // Check signature using etheres tooling
+    let ethers_r = ethers_core::types::U256::from_big_endian(r.to_bytes().as_slice());
+    let ethers_s = ethers_core::types::U256::from_big_endian(s.to_bytes().as_slice());
+    let ethers_v = to_eip155_v(multichain_sig.recovery_id, 31337);
+
+    let signature = ethers_core::types::Signature {
+        r: ethers_r,
+        s: ethers_s,
+        v: ethers_v,
+    };
+
+    let verifying_user_pk = ecdsa::VerifyingKey::from(&user_pk_k256);
+    let user_address_ethers: ethers_core::types::H160 =
+        ethers_core::utils::public_key_to_address(&verifying_user_pk);
+
+    assert!(signature
+        .verify(payload_hash_reversed, user_address_ethers)
+        .is_ok());
+
+    // Check if recovered address is the same as the user address
+    let signature_for_recovery: [u8; 64] = {
+        let mut signature = [0u8; 64]; // TODO: is there a better way to get these bytes?
+        signature[..32].copy_from_slice(&r.to_bytes());
+        signature[32..].copy_from_slice(&s.to_bytes());
+        signature
+    };
+
+    let recovered_from_signature_address_web3 = web3::signing::recover(
+        &payload_hash_reversed,
+        &signature_for_recovery,
+        multichain_sig.recovery_id as i32,
+    )
+    .unwrap();
+    assert_eq!(user_address_from_pk, recovered_from_signature_address_web3);
+
+    let recovered_from_signature_address_ethers = signature.recover(payload_hash_reversed).unwrap();
+    assert_eq!(
+        user_address_from_pk,
+        recovered_from_signature_address_ethers
+    );
+
+    let recovered_from_signature_address_local_function =
+        recover(signature, payload_hash_reversed).unwrap();
```

Comment:
nit: We could delete the local recover() and the as_signature() used for local recover() since they do exactly same thing  as the recover() from the 2 crates, it just allowed me to print things out while debugging.